apps/glusterfs/volume_entry.go

```diff
@@ -140,11 +140,15 @@ func NewVolumeEntryFromRequest(req *api.VolumeCreateRequest) *VolumeEntry {
 
 	if vol.Info.Block {
 		vol.Info.BlockInfo.FreeSize = req.Size
-		// prepend the gluster-block group option,
-		// so that the user-specified options can take precedence
-		vol.GlusterVolumeOptions = append(
-			[]string{"group gluster-block"},
-			vol.GlusterVolumeOptions...)
+
+		if !req.SkipBlockHostingVolumeOptions {
```

Comment:
Yes, I chose this path because even changing the order of options has an effect in gluster. So de duplicating the options using a map(which changes the order) wasn't right.

Reply:
+1 on not using the map, wrt to you comment about ordering. But this seems a bit weird to me in that if we're going to plumb through this skip option we could just plumb through custom block hosting volume options.

Reply:
If I understand this correctly, you mean, we could have another member in volumecreaterequest that holds blockvolumeoptions? How would that be different than GlusterVolumeOptions we have already?

Reply:
Summary of call I had with @phlogistonjohn @jarrpa : 

1. Agreement on separating the block flag to mean "can host blockvolumes" and providing a different way to say what block properties it would have.
2. A double negative in cli is bad, let us not have a Skip..* name in cli
3. But if we introduce an option that is not negative, keeping backward compatibility would me we have to change the bool parameter to a pointer. This would be to workaround the json treats missing bool as default false.
4. There is one way to do it cleanly, where we use different name in cli(like use-default-block-options) and retain the negative name in API(to have backward compatibility).